### ISSUE TITLE
Gradle takes forever to run a task sometimes

### ISSUE THREAD

**Issue by CristianGM** [gradle/gradle#8332]
Since yesterday it happened 5 times that when I run a Gradle task ( both from AS and from cli ) my laptop is working really hard but the task never ends.
After a while I did look on the mac Activity Monitor and discovered a java process was eating 800% of my CPU.
The first time it happened I canceled the job, and tried to run `./gradlew bSPP` but the result have been:
> FAILURE: Build failed with an exception.
> * What went wrong:
> Timeout waiting to lock daemon addresses registry. It is currently in use by another Gradle instance.
> Owner PID: 18318
> Our PID: 24376
> Owner Operation: 
> Our operation: 
> Lock file: /Users/cristiangarcia/.gradle/daemon/5.1.1/registry.bin.lock

The Owner PID was the java process, I killed it and everything went back to normal.

Last builds have been with Kotlin 1.3.20 and parallel tasks enabled, but the first one I'm 99% sure was before 1.3.20 was even announced.

What can I look for to provide more info the next time it happens?

### Your Environment
Gradle 5.1.1
MacBook, Mojave 10.14.2, 32GB Ram
AS 3.3

I have not much more data to share because it happened in different projects

No Scan available

**Comment by eskatos** [comment]
Hi @CristianGM 👋 

If it happens again, run VisualVM (ships with the JDK) and connect to the daemon VM. The CPU graph distinguishes time spent on GC, you should then quickly see if it's a heap pressure issue. In that case you can give the daemon more heap, and, look further at what in your build is requiring that much heap by taking a heap dump and looking into it. If it's not a heap pressure issue, then a thread dump would help finding the culprit.

**Comment by CristianGM** [comment]
Thanks for the suggestion Paul, I'll have it ready.

Last time it happened it was on a smaller project and I waited, the build ends successful, is just taking a looooooong time, turning a 45s build into 15 minutes sometimes.
I have a dump but it's 5GB and I should look how to share it.

Every time it happened I was moving on some scripts for the buildSrc and making them .gradle.kts, or modifying those kts.

I have activated the scans on my machine to see if I catch the next time it happens.

Anyway, it will be tomorrow.
